src/test/scala/org/neo4j/spark/DataSourceReaderWithApocTSE.scala

```diff
@@ -326,6 +326,258 @@ class DataSourceReaderWithApocTSE extends SparkConnectorScalaBaseWithApocTSE {
     assertEquals("P0M17DT43200S", res(1).get(5))
   }
 
+  @Test
+  def testReadNodeWithEqualToFilter(): Unit = {
+    val df: DataFrame = initTest(s"""
+     CREATE (p1:Person {name: 'John Doe'}),
+      (p2:Person {name: 'Jane Doe'})
+     """)
+
+    val result = df.select("name").where("name = 'John Doe'").collectAsList()
+
+    assertEquals(1, result.size())
+    assertEquals("John Doe", result.get(0).getString(0))
+  }
+
+  @Test
+  def testReadNodeWithNotEqualToFilter(): Unit = {
+    val df: DataFrame = initTest(s"""
+     CREATE (p1:Person {name: 'John Doe'}),
+      (p2:Person {name: 'Jane Doe'})
+     """)
+
+    val result = df.select("name").where("NOT name = 'John Doe'").collectAsList()
```

Comment:
is there also a not operator on the predicate API?

Reply:
We added a test with the `!=` operator

Reply:
I meant on the spark filter API i.e. can I do .not().where() ?

Reply:
No you can't